### ISSUE TITLE
Certificates trusted by Mozilla fail the check

### ISSUE THREAD

**Issue by DaKnOb** [gutworth/httpswatch#16]
Sites with certificates trusted by the latest Mozilla Firefox are being reported as not trusted.
More specifically, I have observed such behavior from ```VeriSign Class 3 Public Primary Certification Authority - G5``` and ```AddTrust Public Services Root``` so far. Will update this bug if I find more root CAs included in the ```moz-certs.pem``` and reported as not trusted.

**Comment by gutworth** [comment]
See if `curl` fails, too. Browsers tend to be more aggressive at building trust chains when there are missing intermediate certs.

**Comment by DaKnOb** [comment]
Nope, ```curl (1)``` does not fail on getting the site over ```https```.

**Comment by DaKnOb** [comment]
To make a clarification, I visit the aforementioned websites with Firefox and there is no error but the script reports an error in the certificate and that it's not trusted by Firefox.

**Comment by DaKnOb** [comment]
Update: Two websites that make sure of the same Root CA are being reported one as trusted and the other as not trusted. The site reported as not trusted has an EV Certificate if that helps.

**Comment by gutworth** [comment]
What are the sites?

**Comment by DaKnOb** [comment]
https://www.nbg.gr/
https://www.alpha.gr/

**Comment by gutworth** [comment]
I think we're hitting the recent removal of 1024-bit certs from the Mozilla cert bundle and the fact that OpenSSL doesn't support certificate path discovery. Does it work with my recent change to `moz-certs.pem`?

**Comment by DaKnOb** [comment]
Some sites (mostly EV's) have been fixed. Others unfortunately continue to be reported as not trusted.. I am still looking for a better file / solution to this problem as we speak..

**Comment by DaKnOb** [comment]
I have manually added the Root CA's again into the ```moz-certs.pem``` file and it still cannot mark the websites as trusted. Checked the signature of the certificate presented and the certificate added in the file and they are identical.. Problem with OpenSSL? Something else?

**Comment by gutworth** [comment]
I think fixed https://www.nbg.gr/ hanging.

**Comment by gutworth** [comment]
Both www.nbg.gr and www.alpha.gr work for me now.

**Comment by DaKnOb** [comment]
Yup, they seem to work fine.
The problem now is that the script gets ```EPERM``` ( Issue #27 ) instead of anything else to show a better error message.

**Comment by gutworth** [comment]
So, is this issue resolved?

**Comment by gutworth** [comment]
What domains are still failing?

**Comment by DaKnOb** [comment]
An example is https://www.auth.gr

**Comment by gutworth** [comment]
Evidently that's a [recently fixed](https://rt.openssl.org/Ticket/Display.html?id=3562&user=guest&pass=guest) limitation of OpenSSL.
